### PR TITLE
(HAM-1164) Repetitive components

### PR DESCRIPTION
**Pull request by kbudkaLeanspace**
Please take a really close look ;) I'm doing it for the first time

### CODE REVIEW

File: services/streams/streams/parsers.go [hunk 1]
```diff
@@ -61,6 +65,17 @@ func (streamComp *StreamComponent) ToMap() map[string]any {
 	return streamCompMap
 }
 
+func (repetitive *Repetitive) ToMap() map[string]any {
+	repetitiveMap := make(map[string]any)
+	if repetitive.Value != nil {
```
Comment (jerome-leanspace) [inline services/streams/streams/parsers.go:70]:
```suggestion
	if repetitive!= nil && repetitive.Value != 0 {
```
the default value of an int is 0 and not nil

File: services/streams/streams/parsers.go [hunk 2]
```diff
@@ -61,6 +65,17 @@ func (streamComp *StreamComponent) ToMap() map[string]any {
...
+	    repetitiveMap["value"] = repetitive.Value
+	}
+	if repetitive.Path != nil {
```
Comment (jerome-leanspace) [inline services/streams/streams/parsers.go:73]:
```suggestion
	if repetitive!= nil && repeetitive.Path!="" {
```
the default value of a string is an empty string and not nil

**Review by jerome-leanspace — CHANGES_REQUESTED** [review]
I left some comments
Also since you renamed one field here, it will be a breaking change so we may have to bump the version to 1.0.0 when releasing

**Comment by jerome-leanspace** [comment]
Also there's a mix of tab and spaces, can you format it? It can be done with `go fmt`

File: services/streams/streams/models.go [hunk 3]
```diff
@@ -33,6 +33,7 @@ type StreamComponent struct {
 	Order  int     `json:"order"`
 	Path   string  `json:"path"`
 	Type   string  `json:"type"` // CONTAINER / FIELD / SWITCH
+	Repetitive Repetitive `json:"repetitive,omitempty"`
```
[the remark this answers is not in the document]
Comment (kbudkaLeanspace) [reply services/streams/streams/models.go]: fixed

File: services/streams/streams/parsers.go (hunk 1 above)
Comment (kbudkaLeanspace) [reply services/streams/streams/parsers.go]: fixed

File: services/streams/streams/parsers.go [hunk 4]
```diff
@@ -203,6 +218,11 @@ func (streamComp *StreamComponent) FromMap(streamCompMap map[string]any) error {
 	streamComp.Path = streamCompMap["path"].(string)
 	streamComp.Type = streamCompMap["type"].(string)
 	streamComp.Valid = streamCompMap["valid"].(bool)
+	if len(streamCompMap["repetitive"].([]any)) > 0 {
+		if err := streamComp.Repetitive.FromMap(streamCompMap["repetitive"].([]any)[0].(map[string]any)); err != nil {
```
[the remark this answers is not in the document]
Comment (kbudkaLeanspace) [reply services/streams/streams/parsers.go]: fixed

**Review by jerome-leanspace — APPROVED** [review]
LGTM
just a small comment: you can check the formatting
Then you can discuss how to release this change

**Pull request merged**